### ISSUE TITLE
Remove Unity/Appindicator support?

### ISSUE THREAD

**Issue by brot** [gpodder/gpodder#767]
Hi,
I tried running git master branch on Ubuntu 20.04 and get the following error message for the `ubuntu_appindicator` extension
```
1585397262.349251 [gpodder.extensions] ERROR: Cannot load ubuntu_appindicator from /home/bernd/Development/gpodder/gpodder/share/gpodder/extensions/ubuntu_appindicator.py: cannot import name AppIndicator3, introspection typelib not found
Traceback (most recent call last):
  File "/home/bernd/Development/gpodder/gpodder/src/gpodder/extensions.py", line 260, in set_enabled
    self.load_extension()
  File "/home/bernd/Development/gpodder/gpodder/src/gpodder/extensions.py", line 302, in load_extension
    module_file = imp.load_module(basename, fp, self.filename,
  File "/usr/lib/python3.8/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/lib/python3.8/imp.py", line 171, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 783, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/bernd/Development/gpodder/gpodder/share/gpodder/extensions/ubuntu_appindicator.py", line 8, in <module>
    from gi.repository import AppIndicator3 as appindicator
  File "/usr/lib/python3/dist-packages/gi/importer.py", line 132, in load_module
    raise ImportError('cannot import name %s, '
ImportError: cannot import name AppIndicator3, introspection typelib not found
```
And the following error message for `ubuntu_unity` extension
```
1585397262.353139 [gpodder.extensions] ERROR: Cannot load ubuntu_unity from /home/bernd/Development/gpodder/gpodder/share/gpodder/extensions/ubuntu_unity.py: [Errno 2] No such file or directory: 'python'
Traceback (most recent call last):
  File "/home/bernd/Development/gpodder/gpodder/src/gpodder/extensions.py", line 264, in set_enabled
    self.module.on_load()
  File "/home/bernd/Development/gpodder/gpodder/share/gpodder/extensions/ubuntu_unity.py", line 45, in on_load
    self.process = util.Popen(['python', __file__],
  File "/home/bernd/Development/gpodder/gpodder/src/gpodder/util.py", line 2028, in __init__
    super(Popen, self).__init__(*args, **kwargs)  # Python 2 syntax
  File "/usr/lib/python3.8/subprocess.py", line 854, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/usr/lib/python3.8/subprocess.py", line 1702, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'python'
```
or the following error meesage if i change `python` to `python3`
```
/home/bernd/Development/gpodder/gpodder/share/gpodder/extensions/ubuntu_unity.py:61: PyGIWarning: Unity was imported without specifying a version first. Use gi.require_version('Unity', '7.0') before import to ensure that the right version gets loaded.
  from gi.repository import Unity, GObject
/home/bernd/Development/gpodder/gpodder/share/gpodder/extensions/ubuntu_unity.py:99: PyGIDeprecationWarning: Since version 3.11, calling threads_init is no longer needed. See: https://wiki.gnome.org/PyGObject/Threading
  GObject.threads_init()
/home/bernd/Development/gpodder/gpodder/share/gpodder/extensions/ubuntu_unity.py:100: PyGIDeprecationWarning: GObject.MainLoop is deprecated; use GLib.MainLoop instead
  loop = GObject.MainLoop()
```

In bin/gpodder there is the following check
```
gpodder.ui.unity = (os.environ.get('DESKTOP_SESSION', 'unknown').lower() in
                    ('ubuntu', 'ubuntu-2d'))
```
which assumes that a desktop session "ubuntu" is using unity. But Ubuntu has been using Gnome Shell for some time now and asking for the DESKTOP_SESSION just returns `ubuntu`
```
>>> import os
>>> os.environ.get('DESKTOP_SESSION', 'unknown').lower()
'ubuntu'
```

So could we just remove the `gpodder.ui.unity` flag and the two extensions?
* ubuntu_appindicator
* ubuntu_unity

**Comment by elelay** [comment]
Yes, sure!
MR is welcome :-)

**Comment by elelay** [comment]
From a test with 19.10 in a VM:

    sudo apt install unity-session lightdm libappindicator3-dev

unity is not correctly detected:

    $ python3
    >>> import os
    >>> os.environ.get('XDG_CURRENT_DESKTOP', 'unknown').lower()
    'unity:unity7:ubuntu'
    >>> os.environ.get('DESKTOP_SESSION', 'unknown').lower()
    'unity'

Modifying the condition in `bin/gpodder` activates the `ubuntu_appindicator.py` extension and it works.

ubuntu_unity.py needs further modification (change python to python3 and add encoding='utf-8' in invocation) to work. I've also refactored it to use the gobject bindings directly.

**Comment by elelay** [comment]
On ubuntu 19.10, gnome 3 desktop,

    >>> os.environ.get('DESKTOP_SESSION', 'unknown').lower()
    'ubuntu'
    >>> os.environ.get('XDG_CURRENT_DESKTOP', 'unknown').lower()
    'ubuntu:gnome'

**Comment by elelay** [comment]
Same on ubuntu 18.04 (unity and gnome3)

**Comment by elelay** [comment]
See #776 for fixed detection and code...

**Comment by elelay** [comment]
Closed by #776
